### PR TITLE
Update PID BF code

### PR DESCRIPTION
**Pull request by panjinjin**
added pile-up cut

### CODE REVIEW

**Comment by rbertens** [comment]
+test 
Hi @panjinjin , here's the '+test', but I don't see the necessary changes to the CMakeList.txt file, so I'm not so sure if it will go through ..
GOod night,
REdmer

**Comment by panjinjin** [comment]
Hi, @rbertens ,
Thanks for help! And sorry for the inconvenience! I updated CMakeList.txt by adding the following line to it at line 60. 
set(ALIROOT_DEPENDENCIES ANALYSISalice CORRFW PWGmuon PWGCFCorrelationsBase PWGJETFW OADB)
Now the code should run. Could you please approve the test again?
Thanks!
Au-Au

**Comment by rbertens** [comment]
hi @panjinjin , you still miss one thing in the CMaklists.txt, which is the location of the header file that needs to be include, so you have to add a line '${AliPhysics_SOURCE_DIR}/OADB/' to the list of 'additional includes' (the list starts at line 23 of the CMakeLists.txt file)
Could you add this? Then I can approve the test again. 
My best,
Redmer

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,6 +20,7 @@ endforeach()
 message(STATUS "CMake platform: ${CMAKE_SYSTEM}")
 message(STATUS "Build folder: ${AliPhysics_BINARY_DIR}")
 message(STATUS "Source folder: ${AliPhysics_SOURCE_DIR}")
+message(STATUS "Source folder: ${AliPhysics_SOURCE_DIR}/OADB/")
```
Comment (dberzano) [inline CMakeLists.txt]: Nope. Do not ever change the toplevel `CMakeLists.txt`. Change the closest to your tasks's directory.

**Comment by panjinjin** [comment]
Hi, @rbertens !
Thank you for your information! 
I just added the following line at line 23 of CMakeLists.txt:
message(STATUS "Source folder: ${AliPhysics_SOURCE_DIR}/OADB/")
Is this correct? 
Could you please approve the tests?
Thanks,
Au-Au

**Comment by dberzano** [comment]
@panjinjin I have answered in my review. When you change `CMakeLists.txt`, the approval falls back on the admins and @rbertens cannot approve the test. Anyway this change is wrong :-(

File: CMakeLists.txt [hunk 2]
```diff
@@ -57,7 +56,7 @@ if(CMAKE_INSTALL_PREFIX STREQUAL "${AliPhysics_SOURCE_DIR}")
 endif()
 
 # Path to additonal modules
-set(CMAKE_MODULE_PATH "${AliPhysics_SOURCE_DIR}/cmake")
+set(CMAKE_MODULE_PATH "${AliPhysics_SOURCE_DIR}/cmake" "${AliPhysics_SOURCE_DIR}/OADB")
```
Comment (dberzano) [inline CMakeLists.txt]: Again: you should not touch the toplevel CMakeLists.txt

Comment (panjinjin) [inline CMakeLists.txt]: Hi, @dberzano,
thank you for your information! 
However, I am instructed to add a line '${AliPhysics_SOURCE_DIR}/OADB/' to the list of 'additional includes' (the list starts at line 23 of the CMakeLists.txt file), in order that I could use the AliEventCuts class. So, where should I put the line '${AliPhysics_SOURCE_DIR}/OADB/' in the CMakeLists.txt?
Thanks,
Au-Au

Comment (dberzano) [inline CMakeLists.txt]: As said already, in the CMakeLists.txt closer to the directory `PWGCF/Correlations/DPhi`. This means `PWGCF/Correlations/DPhi/CMakeLists.txt`.

You have to change, as discussed with @mpuccio and @rbertens:

* `ALIROOT_DEPENDENCIES` to add OADB, the new line becoming `set(ALIROOT_DEPENDENCIES ANALYSISalice CORRFW PWGmuon PWGCFCorrelationsBase PWGJETFW OADB)`
* `${AliPhysics_SOURCE_DIR}/OADB` in the `include_directories` statement

**Comment by rbertens** [comment]
Hi @panjinjin ,
The changes should not be made to the top level CMakeLists.txt, but to the one that you can find in the folder where your analysis code is stored, i.e. in PWGCF/Correlations/DPhi/CMakeLists.txt

**Comment by panjinjin** [comment]
Hi, @dberzano !
I changed the top level CMakeLists.txt back to the way it was, though it is still in my pull now ( since I don't know how to remove it from my pull, sorry ). So, is it OK?
Thanks,
Au-Au

**Comment by rbertens** [comment]
+test

**Comment by rbertens** [comment]
Hi @panjinjin  , now the code just does not compile because of a syntax error. I think you have two accidental comma's at lines 428 and 778 of your .cxx, that should be there (' fEventCut(0),'  should be  'fEventCut(0)' ). Did you test if this code compiles for you, before making the pull request .. ?

**Comment by rbertens** [comment]
+test

**Comment by panjinjin** [comment]
Hi, @rbertens !
Thanks for pointing out the bug! It is corrected now. 
Yes, before I made the pull request, the code was running "magically" perfectly locally. 
Sorry for all the mistakes I made here and all the inconveniences I generated.
Cheers,
Au-Au

**Comment by panjinjin** [comment]
Hi, @rbertens !
Thanks for all the help you provided! I really appreciate it!
Finally the code passed the tests.
Onward,
Au-Au

**Comment by rbertens** [comment]
+1 Hi @panjinjin , you're welcome, here's the '+1'. Happy analyzing !

**Pull request merged**